### ISSUE TITLE
[Bug] tsconfig "extends" error

### ISSUE THREAD

**Issue by RebeccaStevens** [webhintio/hint#4715]
# 🐞 Bug report

## Description

In vscode, whenever I set the `"extends"` property in my `tsconfig.json` file, webhint gives the following error message:

`Unexpected token } in JSON at position 705 (typescript-config/is-valid)`

## Details

I'm using the latest vscode extention.
I don't have any custom config (I don't have a `.hintrc` file).
I'm using TypeScript 4.4.2

**Comment by antross** [comment]
Thanks for the report @RebeccaStevens!

I was able to hit the same error in a test project, but I'm not sure if it's for the same reason. In my case the config I was extending had a trailing comma at the end of a list of properties:

```ts
    "compilerOptions": {
        "target": "esnext",
    }
```

Removing the trailing comma fixed the issue.

```ts
    "compilerOptions": {
        "target": "esnext"
    }
```

However this is definitely a bug as it *should* work. Digging in it looks like the problem is that webhint is using a JSONC parser for the top-level tsconfig file, but is handing off to a regular `JSON.parse` for handling extended configs. The latter is much more restrictive on what's allowed (hence the error).

This should probably be fixed by updating [`loadJSONFile`](https://github.com/webhintio/hint/blob/main/packages/utils-fs/src/load-json-file.ts#L7) to use [`parseJSON`](https://github.com/webhintio/hint/blob/9253978337228a09a57b08eb6722c37d93dbb865/packages/utils-json/src/parse-json.ts#L167) (which uses JSONC under the covers) instead of `JSON.parse`.

**Comment by antross** [comment]
@sarvaje just checking, do you see any issues with switching `loadJSONFile` to use the JSONC-backed `parseJSON` helper instead of `JSON.parse`?

**Comment by sarvaje** [comment]
No, if we are already using `loadJSONFile` for the top-level tsconfig, I think it makes sense to use it for the extended configs too.

**Comment by jamclema** [comment]
PR #5031 has made this change to the JSON parsing behavior and there should now be a useful error message when extending from an invalid JSON file.
